### PR TITLE
Documents envoy resource manager runtime flag

### PR DESCRIPTION
**Pull request by ZPain8464**
This PR documents the new Envoy resource manager runtime flag behavior. It also updates the text in other parts of the page for consistency.

@kralicky if you could review for technical accuracy, that would be appreciated. If you find the text isn't clear, or if you have suggestions for improvement, please let me know as well. Thanks!

Resolves https://github.com/pomerium/documentation/issues/1432

### CODE REVIEW

File: content/docs/reference/runtime-flags.md [hunk 1]
```diff
@@ -34,9 +34,10 @@ The available flags are:
...
 | `match_any_incoming_port` | For a route where the From URL does not contain a port number, allow it to match incoming requests with any port number. See the section on [Port matching behavior](/docs/reference/routes/from#port-matching-behavior) for more details. | `true` |
 | `legacy_identity_manager` | The way Pomerium manages IdP session refresh has been newly rewritten in v0.26 for enhanced performance and reliability. When this flag is enabled, Pomerium will revert to the older implementation. | `false` |
+| `envoy_resource_manager_enabled` | Monitors control group (cgroup) memory usage of containerized processes (including Pomerium, Envoy, and client connections and requests) and applies overload actions when memory thresholds are exceeded to reduce memory consumption. See [memory thresholds](#envoy-resource-manager-memory-thresholds) to review thresholds and their corresponding overload actions. | `true` |
```
Comment (kralicky) [inline content/docs/reference/runtime-flags.md:40]: what do you think about this wording instead?

```suggestion
| `envoy_resource_manager_enabled` | Monitors control group (cgroup) memory usage of all processes in the container (including both Pomerium and Envoy) and applies overload actions when memory thresholds are exceeded to reduce memory consumption. See [memory thresholds](#envoy-resource-manager-memory-thresholds) to review thresholds and their corresponding overload actions. | `true` |
```

File: content/docs/reference/runtime-flags.md [hunk 2]
```diff
@@ -48,3 +49,22 @@ runtime_flags:
...
+| **90%** | `shrink_heap` | Envoy will shrink its heap memory every 10 seconds. |
+| **90%** | `reset_high_memory_stream` | Envoy will start resetting streams using the most memory. As memory usage increases, the eligibility threshold is reduced. |
+| **>85%** | `reduce_timeouts` | Envoy will gradually reduce timeouts by up to 50%. |
+| **95%** | `stop_accepting_connections` | Envoy will stop accepting new connections, but keep existing ones open. |
+| **98%** | `disable_http_keepalive` | Envoy will disable HTTP keep-alive, which prevents starting new HTTP/2 streams and cancels existing ones. |
+| **99%** | `stop_accepting_requests` | Envoy will stop accepting all new requests. |
+
+:::note Pod resource limits behavior
+
+If you [specify memory limits](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/) for containers in a Pod, the `envoy_resource_manager_enabled` runtime flag will be **enabled** by default.
```
Comment (kralicky) [inline content/docs/reference/runtime-flags.md:68]:
```suggestion
The `envoy_resource_manager_enabled` runtime flag is set to **true** by default, but only takes effect if you [specify memory limits](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/) for the Pomerium pod.
```

File: content/docs/reference/runtime-flags.md (hunk 1 above)
Comment (ZPain8464) [reply content/docs/reference/runtime-flags.md:40]: @kralicky I like your suggestion, but adjusted the wording slightly. Is this OK?:
```suggestion
| `envoy_resource_manager_enabled` | Monitors control group (cgroup) memory usage of all processes running in a container (including both Pomerium and Envoy) and applies overload actions when memory thresholds are exceeded to reduce memory consumption. See [memory thresholds](#envoy-resource-manager-memory-thresholds) to review thresholds and their corresponding overload actions. | `true` |
```

File: content/docs/reference/runtime-flags.md (hunk 2 above)
Comment (ZPain8464) [reply content/docs/reference/runtime-flags.md]: I like this rewording, too. Done.

File: content/docs/reference/runtime-flags.md (hunk 1 above)
Comment (kralicky) [reply content/docs/reference/runtime-flags.md:40]: The wording "_a container_" seems confusing to me, maybe "running in the container" would work? or "running in a Pomerium container"?

Comment (ZPain8464) [reply content/docs/reference/runtime-flags.md:40]: Let's go with "running in the container".

Comment (ZPain8464) [reply content/docs/reference/runtime-flags.md:40]:
```suggestion
| `envoy_resource_manager_enabled` | Monitors control group (cgroup) memory usage of all processes running in the container (including both Pomerium and Envoy) and applies overload actions when memory thresholds are exceeded to reduce memory consumption. See [memory thresholds](#envoy-resource-manager-memory-thresholds) to review thresholds and their corresponding overload actions. | `true` |
```

**Comment by ZPain8464** [comment]
@kralicky I updated the wording based on your suggestion. Thanks! Do you mind approving when you've had a chance to review?

**Review by kralicky — APPROVED** [review]

**Pull request merged**